pkg/steps/multi_stage/multi_stage.go

```diff
@@ -226,6 +273,30 @@ func (s *multiStageTestStep) Provides() api.ParameterMap {
 }
 func (s *multiStageTestStep) SubTests() []*junit.TestCase { return s.subTests }
 
+// getProfileData fetches the content of the cluster profile secret.
+// This is done both to guarantee it has been correctly imported into the test
+// namespace and to gather information used when generating the test pods.
+func (s *multiStageTestStep) getProfileData(ctx context.Context) error {
+	var secret coreapi.Secret
+	name := s.profileSecretName()
+	key := ctrlruntimeclient.ObjectKey{
+		Namespace: s.jobSpec.Namespace(),
+		Name:      name,
+	}
+	if err := s.client.Get(ctx, key, &secret); err != nil {
+		return fmt.Errorf("could not get cluster profile secret %q: %w", name, err)
+	}
+	if image, ok := secret.Data[vpnImage]; ok {
+		script, ok := secret.Data[vpnScript]
+		if !ok {
+			return fmt.Errorf("VPN script missing in cluster profile %q", name)
+		}
+		s.vpnImage = strings.TrimSpace(string(image))
```

Comment:
For now, this is a free-form `pull spec` added directly to the test container.

In the future, we can restrict it to something more structured.

Reply:
https://issues.redhat.com/browse/DPTP-2852